src/golang/cmd/server/handler/list_workflow_objects.go

```diff
@@ -80,16 +88,160 @@ func (h *ListWorkflowObjectsHandler) Prepare(r *http.Request) (interface{}, int,
-
 	emptyResp := ListWorkflowObjectsResponse{}
 
-	// Get all specs for the workflow.
-	operatorList, err := h.OperatorRepo.GetDistinctLoadOPsByWorkflow(ctx, args.workflowId, h.Database)
+	saveOpList, err := GetDistinctLoadOpsByWorkflow(
+		ctx,
+		args.workflowId,
+		h.OperatorRepo,
+		h.WorkflowDagRepo,
+		h.ArtifactResultRepo,
+		h.Database,
+	)
 	if err != nil {
-		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving workflow objects.")
+		return emptyResp, http.StatusInternalServerError, err
 	}
 
 	return ListWorkflowObjectsResponse{
-		LoadDetails: operatorList,
+		LoadDetails: saveOpList,
 	}, http.StatusOK, nil
 }
+
+// GetDistinctLoadOpsByWorkflow returns a definitive list of all distinct save operators for a given workflow.
+// Fills in any parameterized fields on the operator specs.
+func GetDistinctLoadOpsByWorkflow(
+	ctx context.Context,
+	workflowID uuid.UUID,
+	operatorRepo repos.Operator,
+	workflowDagRepo repos.DAG,
+	artifactResultRepo repos.ArtifactResult,
+	db database.Database,
+) ([]views.LoadOperator, error) {
+	// Get all distinct specs for the workflow.
+	saveOpList, err := operatorRepo.GetDistinctLoadOPsByWorkflow(ctx, workflowID, db)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unexpected error occurred when retrieving workflow objects.")
+	}
+
+	// Now, we check the list of save operators for any that have parameterized table names.
+	// Since these parameterized names are not present on the operator spec, but are instead filled in on the fly at runtime,
+	// we need to fetch those table names from storage. This requires potentially expanding a single save operator into multiple,
+	// each of which represents a unique table name that was saved to.
+	saveOpIDsToExpand := make([]uuid.UUID, 0, len(saveOpList))
+	for _, op := range saveOpList {
+		relationalLoadParams, isRelational := connector.CastToRelationalDBLoadParams(op.Spec.Parameters)
+		if isRelational {
+			if relationalLoadParams.Table == "" {
+				saveOpIDsToExpand = append(saveOpIDsToExpand, op.OperatorID)
+			}
+		}
+	}
+
+	// If there are no parameterized saved table names, continue without expanding the list.
+	if len(saveOpIDsToExpand) == 0 {
+		return saveOpList, nil
+	}
+
+	// Fetch each of the load operators	that need to be expanded.
+	saveOps, err := operatorRepo.GetNodeBatch(ctx, saveOpIDsToExpand, db)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unexpected error occurred when fetching parameterized save operators.")
+	}
+
+	// Use the save operator's ID as the primary key for organization when fetching the artifacts and then their corresponding
+	// artifact results.
+	paramArtifactIDBySaveOpID := make(map[uuid.UUID]uuid.UUID, len(saveOps))
+	for _, saveOp := range saveOps {
+		if len([]uuid.UUID(saveOp.Inputs)) < 2 {
+			return nil, errors.Newf("Expected parameterized save operator %s to have multiple inputs!", saveOp.ID)
+		}
+
+		tableNameParamArtifactID := saveOp.Inputs[0]
```

Comment:
How do we know the parameter artifact is the first input to the save operator?

Reply:
That's just always the order that the save() on the SDK adds it in. I'll add a comment in both places clarifying this assumption.